app/controllers/admin/volunteers.py

```diff
@@ -25,81 +23,90 @@ def getVolunteers(query):
 
     return json.dumps(searchUsers(query))
 
-@admin_bp.route('/event/<eventID>/manage_volunteers', methods=['POST'])
-def updateVolunteerTable(eventID):
-    try:
-        event = Event.get_by_id(eventID)
-    except DoesNotExist as e:
-        print(f"No event found for {eventID}")
-        abort(404)
+@admin_bp.route('/event/<eventID>/manage_volunteers', methods=['GET', 'POST'])
```

Comment:
This might be far fetched but:

I like that the POST and GET are refactored into one controller but I think grouping related functionalities together might be more readable. For instance, placing all the POST request handling logic in a separate function and all GET logic in another and calling both functions in the controller with if statements ` if request.method == "POST":  //call POST function` and same for GET. 
 This will have both POST and GET in one controller but still keeping the logic separated.

Reply:
So like instead of a bunch of function calls in the controller a single logic function for each request that just contains a lot of that stuff inside of it?